builder/build_validate.go

```diff
@@ -348,7 +348,7 @@ func (b *Builder) mcaPkgInfo(manifests []*swupd.Manifest, version, downloadRetri
 
 // resolvePkgFiles queries a package for a list of file metadata.
 func (b *Builder) resolvePkgFiles(pkg *pkgInfo, version int) ([]*fileInfo, error) {
-	queryCmd := "[%{filenames}, %{filesizes}, %{filedigests}, %{filemodes:perms}, %{filelinktos}, %{fileusername}, %{filegroupname}\n]"
+	queryCmd := "[%{filenames}; %{filesizes}; %{filedigests}; %{filemodes:perms}; %{filelinktos}; %{fileusername}; %{filegroupname}\n]"
```

Comment:
I think there should be a FIXME here regarding a future action to eventually replace `;` with a control character. The control characters accepted by rpm's queryformat are the following:

* \a
* \b
* \f
* \n
* \r
* \t
* \v

IMO, \a or \b would be most appropriate, since they are not classified as "space" characters. Ideally, \0 could be used, but queryformat interprets `\0` as `0`, so that's not helpful.

Reply:
I can play around a little bit with these. I chose not to use tabs because it would make Mixer's acceptable file names more restrictive.

Reply:
Yes, that is understandable. However I think the likelihood of `;` appearing in a filename is higher than a literal tab character, so there is a choice of either:

1) using one of the control characters above as the field separator, and also ban that character for all filenames
2) using an already-banned character as the field separator

I would prefer the first approach, but it is debatable.

Another option to avoid the issue entirely would be:

3) reorder the queryformat string so that `%{filenames}` is referenced last. Then you could split each line into a fixed number of fields, and the leftover rightmost field will be `%{filenames}`. Or you could keep the current order, and split from right-to-left. I don't know what's easier with Go :-)

Reply:
>     1. using one of the control characters above as the field separator, and also ban that character for all filenames

The `\a` character is now used for parsing rpm output and has been added to the character blacklist.